### DISCUSSION TITLE
HttpRequestException when sending large payloads

### DISCUSSION THREAD

**Discussion by colin-sim** [dotnet/yarp#1911] [Q&A] [answered]
Hi there,

I'm getting the following error `System.Net.Http.HttpRequestException: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond. (staging-dce.edops.myob.com:443)` when for POST request with large payload. Stacktrace attached below.

The proxy web API application is intended to slowly replace functionality for an existing service. This application has ASP.NET controllers that replace some of the existing service's endpoints. And those that aren't replaced are forwarded to the _old_ service via YARP reverse proxy. 

> info: Yarp.ReverseProxy.Forwarder.HttpForwarder[48]
>       Request: An error was encountered before receiving a response.
>       System.Net.Http.HttpRequestException: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond. (staging-dce.edops.myob.com:443)
>        ---> System.Net.Sockets.SocketException (10060): A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
>          at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
>          at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16 token)
>          at System.Net.Sockets.Socket.<ConnectAsync>g__WaitForConnectWithCancellation|277_0(AwaitableSocketAsyncEventArgs saea, ValueTask connectTask, CancellationToken cancellationToken)
>          at System.Net.Http.HttpConnectionPool.ConnectToTcpHostAsync(String host, Int32 port, HttpRequestMessage initialRequest, Boolean async, CancellationToken cancellationToken)
>          --- End of inner exception stack trace ---
>          at System.Net.Http.HttpConnectionPool.ConnectToTcpHostAsync(String host, Int32 port, HttpRequestMessage initialRequest, Boolean async, CancellationToken cancellationToken)
>          at System.Net.Http.HttpConnectionPool.ConnectAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
>          at System.Net.Http.HttpConnectionPool.AddHttp2ConnectionAsync(HttpRequestMessage request)
>          at System.Threading.Tasks.TaskCompletionSourceWithCancellation`1.WaitWithCancellationAsync(CancellationToken cancellationToken)
>          at System.Net.Http.HttpConnectionPool.GetHttp2ConnectionAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
>          at System.Net.Http.HttpConnectionPool.SendWithVersionDetectionAndRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)
>          at System.Net.Http.DiagnosticsHandler.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
>          at Yarp.ReverseProxy.Forwarder.HttpForwarder.SendAsync(HttpContext context, String destinationPrefix, HttpMessageInvoker httpClient, ForwarderRequestConfig requestConfig, HttpTransformer transformer)

**Comment by Tratcher** [comment, marked as answer]
This is a socket level error, it can't connect to the second app at all. This implies the second app isn't running, or the port is blocked by the firewall, or maybe dns resolved to the wrong IP.

**Reply by colin-sim** [reply]
Thanks @Tratcher. You're right! Looks like the issue was due to my VPN to the service having been terminated.
